hilti/runtime/include/init.h

```diff
@@ -27,6 +27,7 @@ namespace detail {
  * for every module it compiles. */
 struct HiltiModule {
     const char* name{};              /**< name of the HILTI module; for informational purposes */
+    const char* id = nullptr;        /**< unique identifier for the module */
```

Comment:
What do you think about naming this `linker_scope` instead of `id`? That's what's being put in there in the end, but from the current name it's not quite clear where the uniqueness is coming from.

Reply:
I thought about this as well, but arrived at `id` since this seems more generally useful than `linker_scope`. I'd suggest we leave this as is.